ASP.NET API/WAVC_WebApi/Controllers/FriendsManagerController.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using WAVC_WebApi.Data;
+using WAVC_WebApi.Models;
+
+namespace WAVC_WebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [Authorize]
+    [ApiController]
+    public class FriendsManagerController : ControllerBase
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly ApplicationDbContext _dbContext;
+        private readonly FriendsManager _friendsManager;
+        private readonly int _searchResults = 10;
+
+        public FriendsManagerController(UserManager<ApplicationUser> userManager, ApplicationDbContext dbContext)
+        {
+            _userManager = userManager;
+            _dbContext = dbContext;
+            _friendsManager = new FriendsManager(dbContext);
+        }
+
+        [HttpGet]
+        [Route("Friends")]
+        public async Task<List<ApplicationUserModel>> GetFriends()
+        {
+            var sender = await _userManager.GetUserAsync(HttpContext.User);
+            if (sender != null)
+            {
+                return _friendsManager.GetFriends(sender).Select(u => u.ToApplicationUserModel()).ToList();
+            }
+            else
+            {
+                return new List<ApplicationUserModel>();
+            }
+        }
+
+        [HttpGet]
+        [Route("RequestsForUser")]
+        public async Task<List<ApplicationUserModel>> GetRequestsForUser()
```

Comment:
There should be another controller for managing friend requests.
It could look like:
> GET /api/FriendRequest/

Reply:
I don't think 1 method per Controller is the best idea. There will be a ton of file hopping when reading it and setup (usings, class initialization) will add unnecessarily much code.